neuro_san/service/mcp/processors/mcp_tools_processor.py

```diff
@@ -91,11 +97,34 @@ async def call_tool(self, request_id, metadata: Dict[str, Any], tool_name: str,
         input_request: Dict[str, Any] = self._get_chat_input_request(prompt)
         response_text: str = ""
         try:
-            result_generator = service.streaming_chat(input_request, metadata)
-            async for result_dict in result_generator:
-                partial_response: str = await self._extract_tool_response_part(result_dict)
-                if partial_response is not None:
-                    response_text = response_text + partial_response
+            async with asyncio.timeout(self.tool_timeout_seconds):
+                result_generator = service.streaming_chat(input_request, metadata)
+                async for result_dict in result_generator:
+                    partial_response: str = await self._extract_tool_response_part(result_dict)
+                    if partial_response is not None:
+                        response_text = response_text + partial_response
```

Comment:
Bracket streaming results from internal service with asyncio timeout manager.
and handle possible exceptions below similar to what we do in native neuro-san request handler.